pytorch_lightning/utilities/distributed.py

```diff
@@ -381,6 +389,18 @@ def init_dist_connection(
     world_size = world_size if world_size is not None else cluster_environment.world_size()
     os.environ["MASTER_ADDR"] = cluster_environment.master_address()
     os.environ["MASTER_PORT"] = str(cluster_environment.master_port())
+
+    # local rank mapping for device open is needed for hpu devices
+    if torch_distributed_backend == "hcl" or torch_distributed_backend == "hccl":
+        try:
+            import habana_frameworks.torch.core.hccl
+        except Exception:
+            print("hccl backend is not supported, using hcl backend")
+            torch_distributed_backend = "hcl"
+            os.environ["PL_TORCH_DISTRIBUTED_BACKEND"] = "hcl"
+
+        os.environ["ID"] = str(cluster_environment.local_rank())
+
```

Comment:
Sorry, but I don't think this can live here, we are trying everything outside accelerators to be device agnostic whenever possible. So special-casing stuff should be avoided. Even though it feels like overhead, could you maybe subclass this?

Reply:
Makes sense. Will make it redundant and try out